### ISSUE TITLE
Liked Trakt lists only showing duplicates of first 9 entries [BUG]

### ISSUE THREAD

**Issue by SamDW96** [nixgates/plugin.video.seren#302]
**Describe the bug**
In the 'My Liked Movie Lists' section, I see the same first 9 entries from my Trakt account repeated over and over again. My other liked lists are not showing up.

**To Reproduce**
1. Go to 'My Movies'
2. Click on 'My Liked Movie Lists'
3. See error

**Expected behavior**
A list of all my liked lists, with a single entry for each and no duplicates.

**Kodi Version (please complete the following information):**
 - Kodi Version 18.9.0
 - Seren Version 2.0.8

**Additional context**
It seems like the correct amount of liked lists in my Trakt account are listed there, but as stated before, starting from entry #10 it repeats the lists from #1 again. Clearing cache, metadata, rebuilding Trakt database (multiple times) and re-authorizing Trakt has not resolved the issue.

**Comment by nixgates** [comment]
@SamDW96 Sorry did you mean your menu is displaying the correct amount of items but is offering a next button when it shouldn't, or that there are more lists that should be displayed and are not present,

**Comment by SamDW96** [comment]
I mean that there are lists missing. It seems that the total amount of lists in the menu is correct (with the next button in the right place), but the actual lists displayed are just the same first 9 lists repeating a couple times. So say I have 30 liked lists, I only get the first 9 but they are repeated until the total amount is 30. Unfortunately, that means 21 of my lists are not present.

**Comment by nixgates** [comment]
@SamDW96 ok that makes sense, I'll delve into that now and get it fixed

**Comment by nixgates** [comment]
Ok fix identified, will be available in next update

**Comment by nixgates** [comment]
@SamDW96 
Update has been released, please let me know if this fixes your issue!

**Comment by SamDW96** [comment]
Thank you for your work. I'm afraid the bug hasn't been fixed yet. After updating, I still had the same issue. A clean reinstall of Seren with a full data wipe helped initially and I could correctly see all my lists, but when looking back at it now I see the same problem again with the 9 repeating entries. The same happens in the popular and trending lists under 'Discover Movies' by the way. 

Now that I've done a reinstall, I also seem to be unable to authorize RD as mentioned in issue #321.

**Comment by nixgates** [comment]
@SamDW96 can you please try clearing your cache and trying again. I've just checked on my end and can't replicate the problem any more. Not to be painful, but you're definitely on 2.0.9?

**Comment by neogeo71** [comment]
I am on 2.0.9 and can reproduce this issue too. Cleared cache, deauthorize/reauthorized trakt, cleared seren cache, rebuilt tract database etc. Still have the issue...

**Comment by SamDW96** [comment]
Yes, version 2.0.9. Issue still present after clearing cache. I also tried the other things mentioned above (re-auth, rebuilding database, ...), but to no avail.

**Comment by solojer** [comment]
Issue still occurs on 2.0.10.
![seren](https://user-images.githubusercontent.com/77061178/103777350-bd3b4e00-4ffe-11eb-85cb-14e6c3b7b71e.PNG)

**Comment by nixgates** [comment]
@solojer thank you for the image! I can see what you mean now, I had fixed it for lists > page item limit but it appears not for  less than.

Will have it for next update.

**Comment by tayub94** [comment]
Just tested in 2.0.12, issue still remains tried a database rebuild with trakt

**Comment by nixgates** [comment]
@tayub94 Sorry I did say I would but I don't believe I've replicated it yet and that update needed to go out as an emergency :(

**Comment by tayub94** [comment]
No problem man the entire community is eternally grateful for the effort and service you offer 🙏🏾

**Comment by nixgates** [comment]
So I think I've managed to identify the issue, I've tested this against a liked list that only has 2 items and it doesn't happen, but the trending and popular endpoints were still producing the issue.

@tayub94 @SamDW96  not sure if you can, but could I ask one of you to double check my findings above?

If your findings are the same, I've got this fixed for the next update :+1:

**Comment by SamDW96** [comment]
Could you clarify what you mean a little further? Not sure if I get this right. To check, should we populate the Liked Lists section with lists that only have 2 items in them, and remove other lists from it?

**Comment by nixgates** [comment]
@SamDW96  just that to test i only had two liked lists and Seren only displayed those two, and only displayed both of them once.

**Comment by tayub94** [comment]
All good here, tested on shield 2017 and 19, Mac and iPhone, top class as always

**Comment by Njals** [comment]
This issue still persists for me on the following lists; My Liked Shows list, My Liked Movies Lists as well does not show all of my liked shows/movies lists.
Kodi 18.9 / Seren 2.0.16 / Win 10

**Comment by tayub94** [comment]
> This issue still persists for me on the following lists; My Liked Shows list, My Liked Movies Lists as well does not show all of my liked shows/movies lists.
> Kodi 18.9 / Seren 2.0.16 / Win 10

try removing seren and doing a fresh install, failing that try to remove your liked lists and re-like them again

**Comment by Njals** [comment]
> try removing seren and doing a fresh install, failing that try to remove your liked lists and re-like them again

Thank you for the reply I have exhausted this solutions before coming here seeking assistance.

**Comment by rizreflects** [comment]
Same problem for me across multiple devices. When 'paginate lists' is left unchecked, the lists keep repeating themselves. (Tested on windows 10 and CoreELEC)

**Comment by rizreflects** [comment]
im referring to 'my liked lists'.

**Comment by SamDW96** [comment]
> So I think I've managed to identify the issue, I've tested this against a liked list that only has 2 items and it doesn't happen, but the trending and popular endpoints were still producing the issue.
> 
> @tayub94 @SamDW96 not sure if you can, but could I ask one of you to double check my findings above?
> 
> If your findings are the same, I've got this fixed for the next update 👍

Hi there Nix! Sorry to have postponed this so long, but I've finally done some testing today. Like you said, when I removed a bunch of liked lists from my Trakt account so that I only had two left, Seren correctly displayed only these two (after clearing the cache). 

I looked into it further and gradually increased my amount of liked lists to repopulate them (didn't want to lose them), clearing the cache and restarting Kodi after each step. Everything went fine up until I had 10 liked lists, those showed up fine. As soon as I added an 11th, the problem appeared again: suddenly Seren shows 20 items, which were actually the 10 most recently liked lists repeated twice, and with the 'oldest' one now gone. Going any higher, the problem persisted. I have 23 liked lists, which explains why I have this issue while some others don't.

In conclusion, it seems the issue pops up as soon as there is more than 10 items. As soon as you have over 10, Seren starts repeating items, and makes other items inaccessible.

**Comment by drinfernoo** [comment]
Please open this again with a log attached (follow the bug report template) if it is still an issue in 2.0.17.
